src/main/java/io/quarkus/gizmo2/impl/Util.java

```diff
@@ -151,7 +151,9 @@ public static String trackCaller() {
 
     private static String callerOutsideGizmo() {
         return SW.walk(stream -> stream
-                .filter(it -> !it.getClassName().startsWith("io.quarkus.gizmo2") || it.getClassName().endsWith("Test"))
+                .filter(it -> !it.getClassName().startsWith("io.quarkus.gizmo2")
```

Comment:
I didn't think of this before, but it might be slightly more robust to say `! it.getPackageName().equals(...)`. You could then use a `Set` to validate the package name if you wanted to simplify it even further.

Reply:
`StackFrame` doesn't have a `getPackageName()` method. I could use `getDeclaringClass().getPackageName()`, which seems like a longer way of writing what's currently there, but I don't think I could put all package names into a `Set`, because I have to check subpackages as well.

Reply:
Ah, yes you're right. I was thinking of it as `Class`.